File: java/src/main/java/org/example/Sensor.java
package org.example;

import java.util.Random;

public class Sensor {

    private static double OFFSET = 16;

    public double popNextPressurePsiValue()
    {
        double pressureTelemetryValue = samplePressure();
        return OFFSET + pressureTelemetryValue;
    }

    private static double samplePressure()
    {
        // placeholder implementation that simulates a real sensor in a real tire
        Random basicRandomNumbersGenerator = new Random();
        return 6 * basicRandomNumbersGenerator.nextDouble() * basicRandomNumbersGenerator.nextDouble();
    }
}

File: kotlin/src/main/kotlin/org/example/Sensor.kt
package org.example

import kotlin.random.Random

private const val OFFSET = 16.0

class Sensor {

    fun popNextPressurePsiValue(): Double {
        val pressureTelemetryValue = samplePressure()
        return OFFSET + pressureTelemetryValue
    }

    private fun samplePressure(): Double {
        return 6 * Random.nextDouble() * Random.nextDouble()
    }
}